improver/calibration/dataframe_utilities.py

```diff
@@ -443,25 +550,47 @@ def forecast_dataframe_to_cube(
             units=TIME_COORDS["forecast_reference_time"].units,
         )
 
-        for percentile in sorted(df["percentile"].unique()):
-            perc_coord = DimCoord(
-                np.float32(percentile), long_name="percentile", units="%"
-            )
-            perc_df = time_df.loc[time_df["percentile"] == percentile]
+        for var_val in sorted(time_df[representation_type].unique()):
+            var_df = time_df.loc[time_df[representation_type] == var_val]
+            cf_name = var_df["cf_name"].values[0]
+            if representation_type == "threshold":
```

Comment:
As another query related to the "threshold" `representation_type`, I was wondering whether there might be a need, with this representation type for the comparison operator (e.g. "<", ">", or maybe more specifically the options supported in [comparison_operator_dict](https://github.com/metoppv/improver/blob/e9440eeb8ee389d1512d41897801b51d9c73db67/improver/utilities/probability_manipulation.py#L41), etc) to be stored within the dataframe. At the moment, the dataframe would only contain the `forecast` column giving the probability value and the `threshold` column giving the threshold but you wouldn't know how the probability value related to the threshold. Perhaps we should consider whether e.g. `comparison_operator` should be optionally included within the forecast dataframe? @LaurenceBeard may have some thoughts?

Reply:
It would indeed be useful to encode this in the dataframe, but it wouldn't be relevant for the other representations.
How about encoding it in the column name somehow? eg.
```
REPRESENTATION_COLUMNS = [
    "percentile",
    "realization",
    ">threshold",
    "<threshold",
]
```

Reply:
This sounds sensible. We'll move the thresholding into the thresholding functionality and add the two options for threshold into the utilities and test how this works before updating this PR

Reply:
Just to point out, that if taking this approach of augmenting the column name, it would need to support ">", ">=", "<" and "<=" as e.g. >0.0 mm/hr is quite different to >=0 mm/hr for precipitation rate. Alternatively, the `comparison_operator` could be stored in a different column. Would that be better if a table was to contain multiple variables that had been thresholded differently?